### ISSUE TITLE
Correctness testing

### ISSUE THREAD

**Issue by LoicGrobol** [hopsparser/hopsparser#37]
I'd like to have some (unit?) tests to ensure that we are not doing anything wrong, especially as the parser becomes more sophisticated. The current smoke tests ensure that the various commands don't crash but tell us nothing about the correctness of the parser. Some ideas

- Have (slow-running but automated) performance non-regression tests that are run before merging PRs and updating dependency to advert #35-like tragedies in the future
- Ensure that we don't accidentally mix things up when batching, by checking the computational graph with [lightning's gradient trick](https://pytorch-lightning-bolts.readthedocs.io/en/latest/info_callbacks.html#gradient-check-for-batch-optimization)
- …

**Comment by LoicGrobol** [comment]
Idea: add a smoketest that downloads a pretrained model and evaluates it on its dev and test dataset to ensure it gives the same results as when it was trained in the first place. That pretrained model should definitely be cached so deal with #43 first.

**Comment by LoicGrobol** [comment]
We do have some of this and this issue is too vague to be actionable.
